src/main/java/org/advent/dayfour/DayFour.java

```diff
@@ -0,0 +1,84 @@
+package org.advent.dayfour;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class DayFour {
+
+    public void setMatchingNumberPerCard(List<NumbersPerCard> listOfNumbersPerCard) {
+        listOfNumbersPerCard.forEach(n -> n.setMatchingNumbers(totalNumberOfMatchingNumbersPerCard(n)));
+    }
+
+    public void setNumberOfTotalPerCard(List<NumbersPerCard> listOfNumbersPerCard) {
+        for (NumbersPerCard numbersPerCard : listOfNumbersPerCard) {
+            numbersPerCard.addNumberOfTotalCards(1);
+        }
+        for (int i = 0; i < listOfNumbersPerCard.size(); i++) {
+            for (int j = i + 1; j <= i + listOfNumbersPerCard.get(i).getMatchingNumbers(); j++) {
+                int numberOfTotalCards = listOfNumbersPerCard.get(i).getNumberOfTotalCards();
+                listOfNumbersPerCard.get(j).addNumberOfTotalCards(numberOfTotalCards);
+            }
+        }
+    }
+
+    public int totalSumOfNumbersOfTotalPerCard(List<NumbersPerCard> listOfNumbersPerCard) {
+        AtomicInteger totalSum = new AtomicInteger();
+        listOfNumbersPerCard.forEach(n -> totalSum.addAndGet(n.getNumberOfTotalCards()));
+        return totalSum.get();
+    }
+
+    public int totalSumOfAllExponentialValues(List<NumbersPerCard> listOfNumbersPerCard) {
+        AtomicInteger totalValue = new AtomicInteger();
+        listOfNumbersPerCard.forEach(n -> totalValue.addAndGet(exponentialSumOfNumbersMatchingBetweenOnHandAndWinningPerGame(n)));
+        return totalValue.get();
+    }
```

Comment:
curious to hear the reasoning for this implementation

Reply:
Fixed to         return listOfNumbersPerCard.stream().mapToInt(npc -> npc.exponentialSumOfNumbersMatchingBetweenOnHandAndWinningPerGame(npc)).sum();